common/interfaces/user_interface/ui_input_text.c

```diff
@@ -498,11 +515,16 @@ static void next_btn_event_handler(lv_obj_t *next_btn, const lv_event_t event) {
       }
       break;
     case LV_EVENT_CLICKED:
+      /* TODO: Update after refactor*/
+      /* Below statement sets the entered text to a global variable.
+         Correct action: A pointer should be provided by application to
+         fill the entered text to and it should be populated here, and
+         Dummy string should be replaced with user provided pointer */
       mark_input(data->entered_text);
       if (ui_mark_list_choice)
         (*ui_mark_list_choice)(data->current_index + 1);
+      ui_set_text_input_event("Dummy");
```

Comment:
Let's return pointer to global variable (`flow_level.screen_input.input_text`), when the UI component is updated, we will just update the pointer from this point.

Application can rely on the `ui_event.text_ptr` in this case and will not need to update that

Reply:
We can export one more API from UI event getter side to clear the input data for now.